### PR TITLE
Added Remediating modal and changed markdown rendered engine

### PR DESCRIPTION
**Pull request by bond95**
image
image
image

@karelhala @doruskova please, take a look

### CODE REVIEW

File: packages/rule-components/src/ReportDetails/RemediatingModal.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Modal, Button, Tabs, Tab } from '@patternfly/react-core';
+import Markdown from './Markdown';
+
+class RemediatingModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
```
Comment (karelhala) [inline packages/rule-components/src/ReportDetails/RemediatingModal.js:9]: You can write the state directly to the class.
```
class RemediatingModal extends React.Component {
  state = {
    isModalOpen: false,
    activeTabKey: 0
  }
}
```

File: packages/rule-components/src/ReportDetails/RemediatingModal.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          activeTabKey: tabIndex
+      });
+  };
+
+  render() {
+      const { reason, resolution, definitions } = this.props;
+      const { isModalOpen, activeTabKey } = this.state;
+
+      return (
+      <>
```
Comment (karelhala) [inline packages/rule-components/src/ReportDetails/RemediatingModal.js:32]: I don't really like this usage, it is super easy to overlook it. Can you add `Fragment` instead? You can import it with `Component` as well so you don't have to write `React.Fragment`

File: packages/rule-components/src/ReportDetails/Markdown.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import propTypes from 'prop-types';
+
+import doT from 'dot';
+import marked from 'marked';
+import sanitizeHtml from 'sanitize-html';
+
+marked.Renderer.prototype.link = (href, title, text) => {
```
Comment (karelhala) [inline packages/rule-components/src/ReportDetails/Markdown.js:8]: Nice usage of prototype.

File: packages/rule-components/src/ReportDetails/Markdown.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    const externalLinkIcon = '<i class="fas fa-external-link-alt"></i>';
+
+    try {
+        const compiledDot = definitions ? doT.template(template, DOT_SETTINGS)(definitions) : template;
+        const compiledMd = marked(sanitizeHtml(compiledDot, sanitizeOptions));
+
+        return <div dangerouslySetInnerHTML={{
+            __html: compiledMd
+            .replace(/<ul>/gim, `<ul class="pf-c-list" style="font-size: inherit">`)
+            .replace(/<a>/gim, `<a rel="noopener noreferrer" target="_blank">`)
```
Comment (karelhala) [inline packages/rule-components/src/ReportDetails/Markdown.js:41]: There's no href in here. Is that intentional?

**Comment by doruskova** [comment]
@bond95 The remediation modal looks good and can you fix the spacing between links (How to remediate and KCS?)
Snímek obrazovky 2020-05-04 v 8 49 31

Btw. good work!

File: packages/rule-components/src/ReportDetails/Markdown.js (hunk 4 above)
Comment (bond95) [reply packages/rule-components/src/ReportDetails/Markdown.js:41]: I copied it from `inventory-insights`, so I think it should be like this 😃

**Comment by bond95** [comment]
@doruskova unfortunately, I cannot change order, because KB link is part of markdown, so I can add remediation link only after it.

**Comment by doruskova** [comment]
@bond95 And is it possible to have a gap between the description and KCS with Remediation? 
Like this? 
Snímek obrazovky 2020-05-04 v 12 50 21

**Comment by bond95** [comment]
@doruskova ?
image

**Comment by bond95** [comment]
@karelhala updated, please, take a look

**Pull request merged**